interface/modules/custom_modules/oe-module-faxsms/library/rc_sms_notification.php

```diff
@@ -0,0 +1,309 @@
+ * @copyright Unknown
+ * @copyright Copyright (c) 2008 Larry Lart
+ * @copyright Copyright (c) 2018-23 Jerry Padgett
+ * @copyright Copyright (c) 2021 Robert Down <robertdown@live.com>
+ */
+
+//hack add for command line version
+use OpenEMR\Modules\FaxSMS\Controller\AppDispatch;
+
+$_SERVER['REQUEST_URI'] = $_SERVER['PHP_SELF'];
+$_SERVER['SERVER_NAME'] = 'localhost';
+$backpic = "";
+$clientApp = null;
+// for cron
+if (($argc ?? null) > 1 && empty($_SESSION['site_id']) && empty($_GET['site'])) {
+    $c = stripos($argv[1], 'site=');
+    if ($c === false) {
+        echo xlt("Missing Site Id using default") . "\n";
+        $argv[1] = "site=default";
+    }
+    $args = explode('=', $argv[1]);
+    $_GET['site'] = $args[1] ?? 'default';
+}
+if (php_sapi_name() === 'cli') {
+    $_SERVER["HTTP_HOST"] = "localhost";
+    $ignoreAuth = true;
+}
+
+// so service can set some settings if needed on init.
+$sessionAllowWrite = true;
+require_once(__DIR__ . "/../../../../globals.php");
+require_once("$srcdir/appointments.inc.php");
+
+$TYPE = "SMS";
+$CRON_TIME = 150;
+// use service if needed
+if ($TYPE === "SMS") {
+    $clientApp = AppDispatch::getApiService('sms');
+    $cred = $clientApp->getCredentials();
+}
+// close writes
+session_write_close();
+set_time_limit(0);
+
+$SMS_NOTIFICATION_HOUR = $cred['smsHours'];
+$MESSAGE = $cred['smsMessage'];
+
+// check command line for quite option
+$bTestRun = isset($_REQUEST['dryrun']) ? 1 : 0;
+if ($argc > 1 && $argv[2] == 'test') {
+    $bTestRun = 1;
+}
+
+$curr_date = date("Y-m-d");
+$curr_time = time();
+$check_date = date("Y-m-d", mktime((date("h") + $SMS_NOTIFICATION_HOUR), 0, 0, date("m"), date("d"), date("Y")));
+
+//$db_sms_msg = cron_getNotificationData($TYPE);
+$db_sms_msg['sms_gateway_type'] = "SMS";
+$db_sms_msg['message'] = $MESSAGE;
+?>
+    <!DOCTYPE html>
+    <html lang="eng">
+    <head>
+        <title><?php echo xlt("SMS Notification") ?></title>
+    </head>
+    <style>
+      html {
+        font-family: Arial, 'Helvetica Neue', Helvetica, sans-serif;
+        font-size: 14px;
+      }
+    </style>
+    <body>
+        <body>
+            <div>
+                <div>
+                    <p class="text-center"><h2><?php echo xlt("Working and may take a few minutes to finish.") ?></h2></p>
+                </div>
+                <?php
+                if ($bTestRun) {
+                    echo xlt("We are in Test Mode and no reminders will be sent. This test will check what reminders will be sent in when running Live Mode.");
+                }
+                $db_patient = cron_GetAlertPatientData();
+                echo "\n<br>" . xlt('Total of') . ": " . count($db_patient ?? []) . " " . xlt('Reminders Found') . " " . ($bTestRun ? xlt("and will be sending for reminders") . " " : xlt("and Sending for reminders ")) . ' ' . $SMS_NOTIFICATION_HOUR . ' ' . xlt("hrs from now.");
+                ob_flush();
+                flush();
+                // for every event found
```

Comment:
Placeholder for ongoing review
(super neat stuff @sjpadgett ! )

Reply:
btw @sjpadgett , don't worry about all the failed testing on ci. e2e testing has decided to go insane and not an issue with your PR :)